### PR TITLE
basicAuth support

### PR DESCRIPTION
**Pull request by fegu**
Adds http basic authentication support (including docs). Also removes some duplicate entries in the cabal file.
Solves issue #96

### CODE REVIEW

File: IHP/Controller/Session.hs [hunk 1]
```diff
@@ -53,6 +56,12 @@ getSessionUUID name = do
             Just (Just value) -> Just value
             _                 -> Nothing
 
+basicAuth :: (?requestContext::RequestContext) => Text -> Text -> Text -> IO ()
```
Comment (mpscholten) [inline IHP/Controller/Session.hs:59]: Can we add some haddock documentaton `| -- ...` to this function? This is then visible when using e.g. haskell language server.

**Review by mpscholten — COMMENTED** [review]
Thanks, looks good 👍

File: IHP/Controller/Session.hs (hunk 1 above)
Comment (fegu) [reply IHP/Controller/Session.hs:66]: Added now.

Comment (fegu) [reply IHP/Controller/Session.hs:66]: ..and that hunter2 reference still makes me chuckle :)

Comment (mpscholten) [reply IHP/Controller/Session.hs:66]: finally someone got it haha :D

**Review by mpscholten — APPROVED** [review]
Thanks :)

**Pull request merged**